sklego/meta/zero_inflated_regressor.py

```diff
@@ -0,0 +1,122 @@
+
+
+class ZeroInflatedRegressor(BaseEstimator, RegressorMixin):
+    """
+    A meta regressor for zero-inflated datasets, i.e. the targets contain a lot of zeroes.
+
+    `ZeroInflatedRegressor` consists of a classifier and a regressor.
+
+        - The classifier's task is to find of if the target is zero or not.
+        - The regressor's task is to output a (usually positive) prediction whenever the classifier indicates that the there should be a non-zero prediction.
+
+    The regressor is only trained on examples where the target is non-zero, which makes it easier for it to focus.
+
+    At prediction time, the classifier is first asked if the output should be zero. If yes, output zero.
+    Otherwise, ask the regressor for its prediction and output it.
+
+    Parameters
+    ----------
+    classifier : Any, scikit-learn classifier
+        A classifier that answers the question "Should the output be zero?".
+
+    regressor : Any, scikit-learn regressor
+        A regressor for predicting the target. Its prediction is only used if `classifier` says that the output is non-zero.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import ExtraTreesClassifier, ExtraTreesRegressor
+    >>> np.random.seed(0)
+    >>> X = np.random.randn(10000, 4)
+    >>> y = ((X[:, 0]>0) & (X[:, 1]>0)) * np.abs(X[:, 2] * X[:, 3]**2)
+    >>> z = ZeroInflatedRegressor(
+    ... classifier=ExtraTreesClassifier(random_state=0),
+    ... regressor=ExtraTreesRegressor(random_state=0)
+    ... )
+    >>> z.fit(X, y)
+    ZeroInflatedRegressor(classifier=ExtraTreesClassifier(random_state=0),
+                          regressor=ExtraTreesRegressor(random_state=0))
+    >>> z.predict(X)[:5]
+    array([4.91483294, 0.        , 0.        , 0.04941909, 0.        ])
+    """
+
+    def __init__(self, classifier, regressor) -> None:
+        """Initialize."""
+        self.classifier = classifier
+        self.regressor = regressor
+
+    def fit(self, X, y):
+        """
+        Fit the model.
+
+        Parameters
+        ----------
+        X : np.ndarray of shape (n_samples, n_features)
+            The training data.
+
+        y : np.ndarray, 1-dimensional
+            The target values.
+
+        Returns
+        -------
+        ZeroInflatedRegressor
+            Fitted regressor.
+
+        Raises
+        ------
+        ValueError
+            If `classifier` is not a classifier or `regressor` is not a regressor.
+        """
+        X, y = check_X_y(X, y)
+        self._check_n_features(X, reset=True)
+        if not is_classifier(self.classifier):
+            raise ValueError(f"`classifier` has to be a classifier. Received instance of {type(self.classifier)} instead.")
+        if not is_regressor(self.regressor):
+            raise ValueError(f"`regressor` has to be a regressor. Received instance of {type(self.regressor)} instead.")
+
+        self.classifier_ = clone(self.classifier)
```

Comment:
I wonder. What if the user wants to pass a `classifier` that has already been fitted before? Do we really want to refit in these scenarios?

Reply:
True, we could check if the estimators were fitted before. If yes, use them, otherwise proceed as before. But then we could also think about some `refit` keyword. Maybe someone wants to pass in a fitted estimator but wants it to be refit.

Reply:
I added it without the new parameter, maybe it's better not to introduce another parameter.

Reply:
I'm fine with a `refit` parameter. It's also something we have in [other meta models](https://scikit-lego.netlify.app/api/meta.html#sklego.meta.Thresholder).

Reply:
I just struggle a bit with how to implement it. Either as a single parameter for both models, which costs some flexibility, or as two separate parameters such as `refit_classifier` and `refit_regressor`, but this introduces so much parameter overhead. What looks better to you?

Reply:
Ah yeah I see what you mean there. 

I'm fine with forcing both models to be trained for now. We can always add the feature if users really start asking for it. It's not a dealbreaker for me.

Reply:
...to be trained during ZIR fit time, right? :D